fix(mc): Closes #2548 Send a dummy content message to trigger ContentSearch initialization

Mardak opened the pull request:
Fix #2548. @sarracini @k88hudson There must be a better way to do this…?

note: there's two commits with different approaches. The former eagerly inits `ContentSearch` on activity stream init and the latter only inits when the search is rendered/mounted.

k88hudson commented:
@Mardak Personally I prefer the first approach because it keeps the code in a single place and will survive front-end refactors etc, but either seems pretty reasonable.

sarracini commented:
Of the two commits I like the second approach because it's clearer. But really there must be a better way to do this...
Maybe we could change ```ContentSearch``` to listen for an ```init``` message which would call into ```ContentSearch.init()``` and we could send that message on our ```APP_INIT```?
Or maybe it would be simpler to add a guard in ContentSearch's ```init``` and just explicitly call ```ContentSearch.init``` ourselves?

k88hudson commented:
I think it would be good some kind of fix in before https://bugzilla.mozilla.org/show_bug.cgi?id=1363085 to unbreak us

dmose commented:
From what I know now, the guard approach sounds simplest and most contained.

Mardak commented:
> Maybe we could change ```ContentSearch``` to listen for an ```init``` message which would call into ```ContentSearch.init()``` and we could send that message on our ```APP_INIT```?

This is unreliable by itself as `init` will only be called if a page with search is loaded. So in addition to this m-c change, we would need to dispatch the custom event from our page.

> Or maybe it would be simpler to add a guard in ContentSearch's ```init``` and just explicitly call ```ContentSearch.init``` ourselves?

If we were to naively do this, it would be similar to partially backing out the performance gains patch that broke us in the first place albeit just for `ContentSearch`, I would assume this would be `r-`ed. So to make it better, we would still need some way to delay calling `init` (probably on page load) instead of immediately on add-on `INIT`.

Mardak commented:
r?@sarracini I fixed up the tests and cleaned up the awaiter/resolver approach with an inline callback that stashes  the bound callback to be removed in-case it never got called.

as-pine-proxy commented:
[Treeherder: 4f913a2c0652de767e1374b4d1deae3c07e6477a](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=4f913a2c0652de767e1374b4d1deae3c07e6477a)
